### PR TITLE
Configurable log level

### LINKED ISSUE #498: Disable httpx logging?
The logs for my bot are most filled with `INFO` level messages like
```
[03/08/2024 14:32:48][httpx][INFO] HTTP Request: GET https://mattermost.example/api/v4/channels/hdjdsjkjsyhyhiiid932ls7jcz/posts "HTTP/1.1 200 OK"
```

Is there a way to log only errors?

### PR DESCRIPTION
**Pull request by Roy-Orbison**
Backwards-compatible, and only takes precedence over the fallback value.

Closes #498.

This code is untested but also a pretty minor change. I'm happy to modify it.

### CODE REVIEW

**Comment by unode** [comment]
Hi Roy thanks for this PR.

It looks to me that instead of introducing `LOG_LEVEL` with a default of `None` we should instead make it `logging.INFO` and simplify the third line of the code that configures logging (line 86).

I also see that `LOG_LEVEL = logging.DEBUG` might be redundant with or remove the need for `DEBUG = True`.
This is something we could mark as deprecated and a candidate for future removal.
Could you include this in your PR?

`settings.DEBUG` is only used in two locations in the mmpy_bot code so the need for backwards compatibility is slim.

**Comment by Roy-Orbison** [comment]
Yes, it was intended to deprecate `DEBUG`. I didn't expect that many people have it enabled in production, but some might in development, or may uncomment it as needed to diagnose issues, and this shouldn't break those cases. I suppose this would necessitate a minor version bump.

**Comment by Roy-Orbison** [comment]
I also noticed `docs/settings.py` doesn't seem to be referenced anywhere. The ReadTheDocs pages refer to `mmpy_bot/settings.py` instead.

File: docs/settings.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import os
+import logging
```
Comment (unode) [inline docs/settings.py:2]:
```suggestion
```

File: docs/settings.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 
 SECRET_KEY = "b2!=c))9ju1q(=hlxmzb1l+2qbdjn+*cn(5(u@qn4c-0!2v+%v"
-DEBUG = True
+LOG_LEVEL = logging.DEBUG
```
Comment (unode) [inline docs/settings.py:8]:
```suggestion
DEBUG = True
```

File: mmpy_bot/settings.py [hunk 3]
```diff
@@ -91,6 +93,12 @@ def __post_init__(self):
             )
             self.MATTERMOST_API_PATH = self.MATTERMOST_API_PATH[: -len(api_url)]
 
+        if self.DEBUG:
+            warnings.warn(
+                "DEBUG has been deprecated. Use LOG_LEVEL to change verbosity.",
```
Comment (unode) [inline mmpy_bot/settings.py:98]:
```suggestion
                "DEBUG has been deprecated and will be removed in a future release. "
                "Set LOG_LEVEL to logging.DEBUG to increase verbosity.",
```

**Review by unode — DISMISSED** [review]
Please exclude `docs/settings.py` from the PR (I've included suggestions in the review for your convenience) as the DEBUG there is for an unrelated purpose that I believe is, in any case, no longer being used.

**Comment by unode** [comment]
> I also noticed `docs/settings.py` doesn't seem to be referenced anywhere. The ReadTheDocs pages refer to `mmpy_bot/settings.py` instead.

That's a very good point. I don't think there's anything Django specific there as docs are handled by Sphinx but I'll wait for @attzonko's input on it. At birds-eye it looks like some of those files are leftovers from the past and should be removed.

**Comment by attzonko** [comment]
Yeah I have not really poked in the `docs` folder much. I am sure there is some leftovers in there that could be cleaned up.

File: docs/settings.py (hunk 1 above)
Comment (Roy-Orbison) [reply docs/settings.py]: :ok_hand: All suggestions implemented in forced push.

**Comment by unode** [comment]
Seems like black is still failing on code style.
Can you make the linter happy?

**Comment by Roy-Orbison** [comment]
Adjusted using [the playground](https://black.vercel.app/). I think it's slightly less readable, myself, but Black did not like a line length.

File: mmpy_bot/settings.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import collections
 import os
 import warnings
+import logging
```
Comment (unode) [inline mmpy_bot/settings.py:4]:
```suggestion
import logging
import os
import warnings
```

File: tests/unit_tests/bot_test.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 from unittest import mock
 
 import pytest
+import logging
```
Comment (unode) [inline tests/unit_tests/bot_test.py:4]:
```suggestion
import logging
from unittest import mock

import pytest
```

**Comment by unode** [comment]
Merging. There was an integration test failure but I think it was unrelated to the changes.

**Pull request merged**